### PR TITLE
Add Python 3 compatibility

### PR DESCRIPTION
**Pull request by kdious**
Minor changes to make this library python 3 compatible.

### CODE REVIEW

**Comment by kdious** [comment]
I needed to use this library for Python3 but was running into some issues.  I made some minor changes so that it works for Python 3.

This is my first time attempting a port like this so please feel free to provide whatever feedback/comments are necessary.  Hope this can get in so that others will be able to use this nice library for Python 3 projects.

**Comment by kdious** [comment]
👓 @kingthomasc

File: python_purify/core.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def __init__(self, api_key, live=True, rspformat='json', verbose=False,
     @staticmethod
     def _make_options(**kwargs):
         out = []
-        for key, value in kwargs.iteritems():
+        for key, value in kwargs.items():
```
Comment (kdious) [inline python_purify/core.py:48]: https://stackoverflow.com/questions/27580938/python-2-and-3-compatible-way-of-iterating-through-dict-with-key-and-value

File: python_purify/core.py [hunk 2]
```diff
@@ -97,7 +97,7 @@ def _call_method(self, method, **kwargs):
             extra=extra
         )
         if self._verbose:
-            print url
+            print(url)
```
Comment (kdious) [inline python_purify/core.py:106]: https://python-future.org/compatible_idioms.html#print

File: setup.py [hunk 3]
```diff
@@ -1,14 +1,13 @@
-#from distutils.core import setup
```
Comment (kdious) [inline setup.py:1]: Unused

File: python_purify/core.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
 from abc import ABCMeta
-from urllib import quote
-from urllib2 import urlopen
+from six.moves.urllib.parse import quote
```
Comment (kdious) [inline python_purify/core.py:4]: https://stackoverflow.com/questions/31827012/python-importing-urllib-quote

File: python_purify/core.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
...
+from six.moves.urllib.request import urlopen
```
Comment (kdious) [inline python_purify/core.py:5]: https://stackoverflow.com/questions/43578721/mathplotlib-six-moves-urllib-request-import-error

File: setup.py [hunk 6]
```diff
@@ -1,14 +1,13 @@
...
 from os import path
-import setuptools
+from setuptools import setup
 here = path.abspath(path.dirname(__file__))
 
-version_num = "1.2.2"
+version_num = "1.2.3"
```
Comment (kdious) [inline setup.py]: Not sure how you wish to handle versioning.

File: python_purify/__init__.py [hunk 7]
```diff
@@ -1,3 +1,3 @@
-from core import ImagePurify
-from core import WordPurify
-from core import VideoPurify
+from .core import ImagePurify
```
Comment (kdious) [inline python_purify/__init__.py:3]: Without these changes you get the following error:

Traceback (most recent call last):
```
...
...
...
    from python_purify import WordPurify
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/python_purify-1.2.3-py3.6.egg/python_purify/__init__.py", line 1, in <module>
ModuleNotFoundError: No module named 'core'
```

Comment (kdious) [reply python_purify/__init__.py:3]: https://stackoverflow.com/questions/34753206/init-py-cant-find-local-modules/34753235

File: python_purify/tests.py [hunk 8]
```diff
@@ -7,7 +7,7 @@
 bad = 'asdf'
 good = 'That\'s a nice hat.'
 
-jpurify = WordPurify(wordapi, live=False, debug=True)
+jpurify = WordPurify(wordapi, live=False, verbose=True)
```
Comment (kdious) [inline python_purify/tests.py:10]: There is no `debug` flag.  Went with `verbose` instead:
https://github.com/kingthomasc/python-purify/blob/master/python_purify/core.py#L19

Before and after (api key changed to <api_key> in the output):

```
KevinsMacBookPro:python_purify kevin$ python tests.py 
Traceback (most recent call last):
  File "tests.py", line 10, in <module>
    jpurify = WordPurify(wordapi, live=False, debug=True)
TypeError: __init__() got an unexpected keyword argument 'debug'


KevinsMacBookPro:python_purify kevin$ python tests.py 
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.addtoblacklist&api_key=<api_key>&format=json&word=asdf&ds=1
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.getblacklist&api_key=<api_key>&format=json&ds=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.addtowhitelist&api_key=<api_key>&format=json&word=asdfasdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.getwhitelist&api_key=<api_key>&format=json
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.check&api_key=<api_key>&format=json&text=asdf&slink=0&semail=0&rsp=0&sphone=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.check&api_key=<api_key>&format=json&text=That%27s%20a%20nice%20hat.&slink=0&semail=0&rsp=0&sphone=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.checkcount&api_key=<api_key>&format=json&text=asdf%20That%27s%20a%20nice%20hat.%20asdf&slink=0&semail=0&rsp=0&sphone=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.checkcount&api_key=<api_key>&format=json&text=asdfasdf&slink=0&semail=0&rsp=0&sphone=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.return&api_key=<api_key>&format=json&text=asdf%20That%27s%20a%20nice%20hat.&slink=0&semail=0&rsp=0&sphone=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.replace&api_key=<api_key>&format=json&sphone=0&text=asdf%20That%27s%20a%20nice%20hat.&slink=0&semail=0&replacesymbol=8&rsp=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.removefromwhitelist&api_key=<api_key>&format=json&word=asdfasdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.checkcount&api_key=<api_key>&format=json&text=asdfasdf&slink=0&semail=0&rsp=0&sphone=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.removefromblacklist&api_key=<api_key>&format=json&word=asdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.check&api_key=<api_key>&format=json&text=asdf&slink=0&semail=0&rsp=0&sphone=0

KevinsMacBookPro:python_purify kevin$ python3 tests.py 
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.addtoblacklist&api_key=<api_key>&format=json&ds=1&word=asdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.getblacklist&api_key=<api_key>&format=json&ds=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.addtowhitelist&api_key=<api_key>&format=json&word=asdfasdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.getwhitelist&api_key=<api_key>&format=json
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.check&api_key=<api_key>&format=json&semail=0&sphone=0&slink=0&rsp=0&text=asdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.check&api_key=<api_key>&format=json&semail=0&sphone=0&slink=0&rsp=0&text=That%27s%20a%20nice%20hat.
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.checkcount&api_key=<api_key>&format=json&semail=0&sphone=0&slink=0&rsp=0&text=asdf%20That%27s%20a%20nice%20hat.%20asdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.checkcount&api_key=<api_key>&format=json&semail=0&sphone=0&slink=0&rsp=0&text=asdfasdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.return&api_key=<api_key>&format=json&semail=0&sphone=0&slink=0&rsp=0&text=asdf%20That%27s%20a%20nice%20hat.
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.replace&api_key=<api_key>&format=json&semail=0&sphone=0&slink=0&rsp=0&replacesymbol=8&text=asdf%20That%27s%20a%20nice%20hat.
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.removefromwhitelist&api_key=<api_key>&format=json&word=asdfasdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.checkcount&api_key=<api_key>&format=json&semail=0&sphone=0&slink=0&rsp=0&text=asdfasdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.removefromblacklist&api_key=<api_key>&format=json&word=asdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.check&api_key=<api_key>&format=json&semail=0&sphone=0&slink=0&rsp=0&text=asdf

```

**Comment by kd2718** [comment]
Thanks for doing this. What version of python are you running? Have you tested to see if 2.7 still works with all the changes?

**Comment by kdious** [comment]
Python 2.7.10 and Python 3.6.5.

Yes I tested python 2.7 - see below (the tests pass).  I also tested in the app that I am using.

```
KevinsMacBookPro:python_purify kevin$ python3
Python 3.6.5 (v3.6.5:f59c0932b4, Mar 28 2018, 03:03:55) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> exit()
...
...
...
KevinsMacBookPro:python_purify kevin$ python
Python 2.7.10 (default, Oct  6 2017, 22:29:07) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> exit()
...
...
...
KevinsMacBookPro:python-purify kevin$ sudo python setup.py install
running install
Checking .pth file support in /Library/Python/2.7/site-packages/
/usr/bin/python -E -c pass
TEST PASSED: /Library/Python/2.7/site-packages/ appears to support .pth files
...
...
...
Installed /Library/Python/2.7/site-packages/python_purify-1.2.3-py2.7.egg
Processing dependencies for python-purify==1.2.3
Finished processing dependencies for python-purify==1.2.3
...
...
...
KevinsMacBookPro:python-purify kevin$ sudo python3 setup.py install
running install
running bdist_egg
running egg_info
...
...
...
Finished processing dependencies for python-purify==1.2.3
...
...
...
KevinsMacBookPro:python_purify kevin$ python2.7 tests.py 
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.addtoblacklist&api_key=<api_key>&format=json&word=asdf&ds=1
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.getblacklist&api_key=<api_key>&format=json&ds=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.addtowhitelist&api_key=<api_key>&format=json&word=asdfasdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.getwhitelist&api_key=<api_key>&format=json
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.check&api_key=<api_key>&format=json&text=asdf&slink=0&semail=0&rsp=0&sphone=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.check&api_key=<api_key>&format=json&text=That%27s%20a%20nice%20hat.&slink=0&semail=0&rsp=0&sphone=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.checkcount&api_key=<api_key>&format=json&text=asdf%20That%27s%20a%20nice%20hat.%20asdf&slink=0&semail=0&rsp=0&sphone=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.checkcount&api_key=<api_key>&format=json&text=asdfasdf&slink=0&semail=0&rsp=0&sphone=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.return&api_key=<api_key>&format=json&text=asdf%20That%27s%20a%20nice%20hat.&slink=0&semail=0&rsp=0&sphone=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.replace&api_key=<api_key>&format=json&sphone=0&text=asdf%20That%27s%20a%20nice%20hat.&slink=0&semail=0&replacesymbol=8&rsp=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.removefromwhitelist&api_key=<api_key>&format=json&word=asdfasdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.checkcount&api_key=<api_key>&format=json&text=asdfasdf&slink=0&semail=0&rsp=0&sphone=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.removefromblacklist&api_key=<api_key>&format=json&word=asdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.check&api_key=<api_key>&format=json&text=asdf&slink=0&semail=0&rsp=0&sphone=0
...
...
...
KevinsMacBookPro:python_purify kevin$ python3 tests.py 
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.addtoblacklist&api_key=<api_key>&format=json&ds=1&word=asdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.getblacklist&api_key=<api_key>&format=json&ds=0
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.addtowhitelist&api_key=<api_key>&format=json&word=asdfasdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.getwhitelist&api_key=<api_key>&format=json
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.check&api_key=<api_key>&format=json&semail=0&sphone=0&slink=0&rsp=0&text=asdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.check&api_key=<api_key>&format=json&semail=0&sphone=0&slink=0&rsp=0&text=That%27s%20a%20nice%20hat.
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.checkcount&api_key=<api_key>&format=json&semail=0&sphone=0&slink=0&rsp=0&text=asdf%20That%27s%20a%20nice%20hat.%20asdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.checkcount&api_key=<api_key>&format=json&semail=0&sphone=0&slink=0&rsp=0&text=asdfasdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.return&api_key=<api_key>&format=json&semail=0&sphone=0&slink=0&rsp=0&text=asdf%20That%27s%20a%20nice%20hat.
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.replace&api_key=<api_key>&format=json&semail=0&sphone=0&slink=0&rsp=0&replacesymbol=8&text=asdf%20That%27s%20a%20nice%20hat.
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.removefromwhitelist&api_key=<api_key>&format=json&word=asdfasdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.checkcount&api_key=<api_key>&format=json&semail=0&sphone=0&slink=0&rsp=0&text=asdfasdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.removefromblacklist&api_key=<api_key>&format=json&word=asdf
http://api1.webpurify.com/services/rest/?method=webpurify.sandbox.check&api_key=<api_key>&format=json&semail=0&sphone=0&slink=0&rsp=0&text=asdf 
```

File: python_purify/__init__.py (hunk 7 above)
Comment (kd2718) [reply python_purify/__init__.py:3]: please add `from __future__ import absolute_import` to the top of this file. should work for py2 and py3

File: python_purify/core.py (hunk 2 above)
Comment (kd2718) [reply python_purify/core.py:106]: please add `from __future__ import print_function` so we don't loose anything with py2

File: setup.py (hunk 6 above)
Comment (kd2718) [reply setup.py]: adding python 3 support is prettty big. might as well make it 2.0.0.

**Comment by kd2718** [comment]
This is a very through PR. thanks for working on this. I only had a few comments. Take a look and let me know if they cause any issues.

Thanks again!

File: CHANGELOG [hunk 9]
```diff
@@ -1,4 +1,7 @@
 === 1.2.0
 * added CHANGELOG
 * added usehttps to select protocol on class initialization
-* added VideoPurify
\ No newline at end of file
+* added VideoPurify
+
+=== 1.2.3
```
Comment (kd2718) [inline CHANGELOG]: change to 2.0.0

**Comment by kdious** [comment]
Changes made.  Re-ran the tests for both python2 and python3 and they worked fine.

Thanks for creating such a well written library!  The substantive changes were very few so this was easy to port.

**Review by kd2718 — APPROVED** [review]

**Comment by kd2718** [comment]
@kdious I am going to merge this. However, I won't have time to update pypi until later.

**Comment by kd2718** [comment]
@kdious updated on pypi

**Comment by kdious** [comment]
Just tried it.  Works great!  See below (api key hidden)

```
KevinsMacBookPro:text_sanitizer kevin.dious$ pip install python-purify
Collecting python-purify
  Downloading https://files.pythonhosted.org/packages/93/07/7d2a4b9ffeab42c578861da25de3894363c93c68bb29b2b67b91266e75d2/python_purify-2.0.0-py3-none-any.whl
Installing collected packages: python-purify
Successfully installed python-purify-2.0.0
You are using pip version 18.0, however version 19.2.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
KevinsMacBookPro:text_sanitizer kevin$ python3
Python 3.6.5 (v3.6.5:f59c0932b4, Mar 28 2018, 03:03:55) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import python_purify
>>> from python_purify import WordPurify
>>> web_purify_api_key = '<api_key>'
>>> webpurify_text_client = WordPurify(web_purify_api_key)
>>> response = webpurify_text_client.replace('screw you!!', rsp=1)
>>> response['rsp']['text']
'screw you!!'
>>> 
```

**Pull request merged**